Add "user" to Breadcrumb.Type

NLthijs48 opened the pull request:
Version 8.0.0 changed the breadcrumb type from a string into an enum (which I fully support), but it does not include a type called "user". The reason I think there should be one is that the web interface defines an icon for it (and that I was using it in a Sentry client):

![user_breadcrumb](https://cloud.githubusercontent.com/assets/6951068/24831553/0e0e1492-1c9c-11e7-9fff-87d628322754.png)

Let me know if you agree with this addition.

bretthoerner commented:
Thanks! Looks good, the javadoc needs a period so checks don't fail and we need to update the `CHANGES` file. I can do both soon if you don't get to it.

NLthijs48 commented:
I updated the javadoc and added a line to the CHANGES file (of course edit the message if you need to).

I tried building the project before submitting the pull request, but it fails: https://pastebin.com/06pgbb8b. This is on Windows 10 using Intellij IDEA, running `mvn package`. Does the build not work on Windows or is there something wrong about my setup?

bretthoerner commented:
Hmm, I haven't tried it on Windows, I'll have to check it out.

bretthoerner commented:
Thanks, I merged here: https://github.com/getsentry/raven-java/commit/67221eb2d0f3306a2fcec857b7ebff02be692ec5

And made an issue to test/fix Windows: https://github.com/getsentry/raven-java/issues/371
